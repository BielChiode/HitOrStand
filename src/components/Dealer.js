import React from 'react'
import { Image, StyleSheet, Text, View } from 'react-native'
import PropTypes from 'prop-types'
import sumPoints from '../utils/sumPoints'
import { Icon } from 'react-native-elements'

function Dealer({ dealer, stand }) {
  const points = sumPoints(dealer)
  return (
    <View style={styles.dealerContainer}>
      <View style={styles.dealerCardsContainer}>
        {dealer &&
          dealer.map((card, index) =>
            index == 0 ? (
              <Image
                key={index}
                style={styles.cardImage}
                source={{
                  uri: stand
                    ? card.image
                    : 'https://deckofcardsapi.com/static/img/back.png'
                }}
              />
            ) : (
              <Image
                key={index}
                style={[styles.cardImage, { marginLeft: -50 }]}
                source={{
                  uri: card.image
                }}
              />
            )
          )}
      </View>
      <View style={styles.dealerAvatar}>
        {stand && <Text style={styles.textPoints}>Pontos: {points}</Text>}
        <Icon
          raised
          name="user"
          type="font-awesome"
          color="#E84233"
        />
      </View>
    </View>
  )
}

Dealer.propTypes = {
  dealer: PropTypes.array.isRequired,
  stand: PropTypes.bool.isRequired
}

export default Dealer

const styles = StyleSheet.create({
  dealerContainer: {
    position: 'absolute',
    top: 0
  },
  dealerCardsContainer: {
    width: '100%',
    marginTop: 16,
    flexDirection: 'row',
    justifyContent: 'flex-end'
  },
  cardImage: {
    width: 110,
    height: 150
  },
  dealerAvatar: {
    marginTop: 10,
    alignItems: 'center',
    justifyContent: 'flex-end',
    flexDirection: 'row'
  },
  textPoints: {
    fontSize: 20,
    fontWeight: '600',
    marginLeft: 10,
    color: '#FFF'
  }
})
