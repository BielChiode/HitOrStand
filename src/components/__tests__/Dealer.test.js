import { render } from '@testing-library/react-native'
import Dealer from '../Dealer'

describe('Dealer', () => {
  const dealer = [
    {
      code: '5D',
      image: 'https://deckofcardsapi.com/static/img/5D.png',
      suit: 'DIAMONDS',
      value: '5'
    },
    {
      code: '4C',
      image: 'https://deckofcardsapi.com/static/img/4C.png',
      suit: 'CLUBS',
      value: '4'
    }
  ]
  it('Deveria renderizar corretamente com as cartas e uma carta virada para baixo', () => {
    const { toJSON } = render(<Dealer dealer={dealer} stand={false} />)
    expect(toJSON()).toMatchSnapshot()
  })
  it('Deveria renderizar corretamente com as cartas viradas para cima e com pontos', () => {
    const { toJSON, getByText } = render(
      <Dealer dealer={dealer} stand={true} />
    )
    expect(getByText('Pontos: 9'))
    expect(toJSON()).toMatchSnapshot()
  })
})
