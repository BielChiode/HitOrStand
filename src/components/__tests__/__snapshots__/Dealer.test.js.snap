// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dealer Deveria renderizar corretamente com as cartas e uma carta virada para baixo 1`] = `
<View
  style={
    {
      "position": "absolute",
      "top": 0,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "marginTop": 16,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://deckofcardsapi.com/static/img/back.png",
        }
      }
      style={
        {
          "height": 150,
          "width": 110,
        }
      }
    />
    <Image
      source={
        {
          "uri": "https://deckofcardsapi.com/static/img/4C.png",
        }
      }
      style={
        [
          {
            "height": 150,
            "width": 110,
          },
          {
            "marginLeft": -50,
          },
        ]
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        {
          "borderRadius": 28,
          "height": 52,
          "margin": 7,
          "overflow": "hidden",
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 52,
        }
      }
    >
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderRadius": 28,
              "height": 52,
              "justifyContent": "center",
              "width": 52,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#E84233",
                  "fontSize": 24,
                },
                {
                  "backgroundColor": "transparent",
                },
                {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
            testID="iconIcon"
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Dealer Deveria renderizar corretamente com as cartas viradas para cima e com pontos 1`] = `
<View
  style={
    {
      "position": "absolute",
      "top": 0,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "marginTop": 16,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://deckofcardsapi.com/static/img/5D.png",
        }
      }
      style={
        {
          "height": 150,
          "width": 110,
        }
      }
    />
    <Image
      source={
        {
          "uri": "https://deckofcardsapi.com/static/img/4C.png",
        }
      }
      style={
        [
          {
            "height": 150,
            "width": 110,
          },
          {
            "marginLeft": -50,
          },
        ]
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFF",
          "fontSize": 20,
          "fontWeight": "600",
          "marginLeft": 10,
        }
      }
    >
      Pontos: 
      9
    </Text>
    <View
      style={
        {
          "borderRadius": 28,
          "height": 52,
          "margin": 7,
          "overflow": "hidden",
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 52,
        }
      }
    >
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderRadius": 28,
              "height": 52,
              "justifyContent": "center",
              "width": 52,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#E84233",
                  "fontSize": 24,
                },
                {
                  "backgroundColor": "transparent",
                },
                {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
            testID="iconIcon"
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
